src/components/CodeExample.js

```diff
@@ -248,38 +241,27 @@ const CodeSnippet = ({ codeSnippets, title, href }) => {
             )}
           </LangOptionEl>
           <DividerEl />
-          <CopyToClipboard
-            text={SelectedSnippetStr && SelectedSnippetStr}
-            onCopy={() => !isCopied && setCopy(true)}
-          >
-            <CopyIconWrapper
-              ref={CopyToClipboardRef}
-              onMouseEnter={() => {
-                setHover(true);
-              }}
-              onMouseLeave={() => {
-                setHover(false);
-              }}
+          <Tooltip message={isCopied ? "Copied" : "Click to copy"}>
+            <CopyToClipboard
+              text={SelectedSnippetStr}
+              onCopy={() => !isCopied && setCopy(true)}
             >
-              <CopyIcon />
-            </CopyIconWrapper>
-          </CopyToClipboard>
+              <CopyWrapperEl>
+                <CopyIcon />
+              </CopyWrapperEl>
+            </CopyToClipboard>
+          </Tooltip>
```

Comment:
By wrapping the hover target, I could move the position detection logic inside `Tooltip`. The logic is basically the same, but now it's contained within `Tooltip`, which is nice.